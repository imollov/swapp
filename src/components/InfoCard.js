import React from 'react'
import { Card, Heading, Box, Image } from 'rebass/styled-components'
import Details from './Details'

export default ({ img, title, data, ...rest }) => (
  <Card {...rest}>
    <Box p={3}>
      <Heading variant="h3" pb={2} textAlign="center">
        {title}
      </Heading>
      <Image src={img} width={1} pb={3} />
      <Details data={data} />
    </Box>
  </Card>
)
