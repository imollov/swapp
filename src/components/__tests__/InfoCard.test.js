import React from 'react'
import { shallow } from 'enzyme'

import InfoCard from '../InfoCard'

describe('InfoCard Component', () => {
  it('should match snapshot', () => {
    const details = [{ field: 'foo', value: 'bar' }]
    const wrapper = shallow(
      <InfoCard img="test.png" title="title" data={details} />,
    )

    expect(wrapper).toMatchSnapshot()
  })
})
