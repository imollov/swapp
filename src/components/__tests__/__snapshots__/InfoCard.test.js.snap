// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard Component should match snapshot 1`] = `
<ForwardRef>
  <styled.div
    p={3}
  >
    <ForwardRef
      pb={2}
      textAlign="center"
      variant="h3"
    >
      title
    </ForwardRef>
    <ForwardRef
      pb={3}
      src="test.png"
      width={1}
    />
    <DetailList
      data={
        Array [
          Object {
            "field": "foo",
            "value": "bar",
          },
        ]
      }
    />
  </styled.div>
</ForwardRef>
`;
